### ISSUE TITLE
Ability to download single episode or season

### ISSUE THREAD

**Issue by greggameplayer** [Fallenbagel/jellyseerr#264]
### Description

In the current version, it's not possible to only ask for a particular episode or a particular season of a serie.

### Desired Behavior

Be able to ask by episodes and/or by seasons with kinda the same system already in place for film or whole series.

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

**Comment by Fallenbagel** [comment]
Need some clarification on what you mean by it's not possible to request for a particular season of a series as it has always been possible. Just not individual episodes
![Screenshot_20221204_055758_Chrome](https://user-images.githubusercontent.com/98979876/205468699-4d90570f-0d1c-46c7-90a3-06b4904d58da.jpg)

**Comment by greggameplayer** [comment]
Oh okay i haven't seen that when you click on ask it would prompt you that, i thought it would download the entire serie

**Comment by lucasshiva** [comment]
I think requesting individual episodes would be very helpful, especially when you're looking for something new to watch and don't want to request entire seasons just to watch a single episode.

**Comment by ojmaster** [comment]
Is this going to be done?

**Comment by ndom91** [comment]
Saw this post about why it might be hard (https://github.com/Fallenbagel/jellyseerr/issues/592#issuecomment-1847836482).

But Sonarr does implement multiple monitoring options like "Pilot Episode Only". I'm not familiar with the Jellyseerr implementation, how it submits new series / movies Sonarr/Radarr, but I can imagine adding a series as "paused", changing it to "monitor only pilot episode" and then "unpause" it could work.. :thinking:

**Comment by t-pierre** [comment]
Would it be possible to add this function initially for series where sonarr and tmdb agree on episode numbering? And/or perhaps set this feature as experimental, to be activated in the settings if we understand the risks ?

**Comment by Bjarne-Kinkel** [comment]
Is anybody working on this?

**Comment by mrmuiz** [comment]
Still one of the best ideas. I keep wishing.....

**Comment by Gliese-667** [comment]
I think a good starting point would be to show which episodes within a season are currently missing, unless that's already a feature and I'm just totally missing it? Right now it's a little bit of extra effort to do a manual comparison to see if I'm missing an episode in a partially available season, or otherwise I have to pull up the missing episode plugin on the media server

**Comment by totza2010** [comment]
Personally I use sonarr only to check for missing TV episodes, nothing else, and I have a huge problem with TV episodes being different between tmdb and tvdb.
It would be great if jellyseerr could check for missing episodes.

**Comment by Tredecate** [comment]
I'm honestly a little surprised this isn't a thing yet. I just dove into the *arr ecosystem for the first time and everything has been great, _except_ for this missing feature that makes it just a little bit harder to conserve disk space.

So far I've been manually deleting requests for seasons and just launching an automatic search in Sonarr for the first few episodes. Works in a pinch, but feels archaic compared to how well everything else works together!

**Comment by fallenbagel** [comment]
> I'm honestly a little surprised this isn't a thing yet. 

The reason is the discrepancy between tmdb and tvdb when it comes to episode numbering and season ordering.

**Comment by Tredecate** [comment]
> The reason it is hard to implement is the discrepancy between tmdb and tvdb when it comes to episode numbering and season ordering.

I've done a little bit of research into the issue, but I'd appreciate some help with understanding how it realistically affects implementation attempts.

It seems TVDb and TMDb are two entirely separate metadata sources, both driven by community contributions. In some cases, there are disagreements on episode ordering and organization.

The problem isn't due to a technical difficulty, but rather stems from dealing with a situation where Seerr requests "TV Show X, Season 1, Episode 12" (as defined in TMDb) and Sonarr replies "That doesn't exist" (as defined in TVDb) or ends up downloading the wrong episode entirely because TVDb has a different episode listed under S01E12.

Is that the entire problem? Or am I missing something that makes things extra spicy?

**Comment by gauthier-th** [comment]
> > The reason it is hard to implement is the discrepancy between tmdb and tvdb when it comes to episode numbering and season ordering.
> 
> I've done a little bit of research into the issue, but I'd appreciate some help with understanding how it realistically affects implementation attempts.
> 
> It seems TVDb and TMDb are two entirely separate metadata sources, both driven by community contributions. In some cases, there are disagreements on episode ordering and organization.
> 
> The problem isn't due to a technical difficulty, but rather stems from dealing with a situation where Seerr requests "TV Show X, Season 1, Episode 12" (as defined in TMDb) and Sonarr replies "That doesn't exist" (as defined in TVDb) or ends up downloading the wrong episode entirely because TVDb has a different episode listed under S01E12.
> 
> Is that the entire problem? Or am I missing something that makes things extra spicy?

Yes this is the issue. TMDB and TVDB are 2 differents database that have nothing in common. We use TMDB on Jellyseerr, but Sonarr only accepts TVDB (implementation of TMDB has been rejected). Both are completely different database, so it's hard to make both match reliably when the metadata is not arranged in the same way.

**Comment by raedmansour** [comment]
I just wanted to add my support to this feature.
I often need only one or a few episodes, but currently Jellyseerr forces me to request the full season, which leads to unnecessary downloads and storage.

- A "Request Episode" button next to each episode.
- The ability to select multiple specific episodes using checkboxes.
- And still keep the existing "Request Season" option for those who want it.

Thanks for all your great work — jellyseerr is awesome and this feature would make it even better! 🙌

**Comment by orenhecht** [comment]
Shamed to say I just learned about Jellyseerr as I am a Plex and Overseerr user. Anyway, I am adding support for individual episode requests in Overseerr, if it won't be too much of a hassle will try to also add it to jellyseerr.
You can get a glimpse here https://github.com/sct/overseerr/issues/342#issuecomment-3330271059

**Comment by fallenbagel** [comment]
> Shamed to say I just learned about Jellyseerr as I am a Plex and Overseerr user. Anyway, I am adding support for individual episode requests in Overseerr, if it won't be too much of a hassle will try to also add it to jellyseerr.
> You can get a glimpse here https://github.com/sct/overseerr/issues/342#issuecomment-3330271059 

We already have preliminary support for it (just availability check for now)
#1671
We have tvdb support and current plans are episode request to only be enabled when tvdb is enabled. You're more than welcome to add it to jellyseerr with the tvdb integration in mind c:

Episode availability is also enabled like that rn

**Comment by gauthier-th** [comment]
> You're more than welcome to add it to jellyseerr with the tvdb integration in mind c:

FYI our current plans where to make the single episode request available only if TVDB support is enabled for series

**Comment by fallenbagel** [comment]
> > You're more than welcome to add it to jellyseerr with the tvdb integration in mind c:
> 
> FYI our current plans where to make the single episode request available only if TVDB support is enabled for series

That's what I just said 🗿

**Comment by gauthier-th** [comment]
> > > You're more than welcome to add it to jellyseerr with the tvdb integration in mind c:
> > 
> > 
> > FYI our current plans where to make the single episode request available only if TVDB support is enabled for series
> 
> That's what I just said 🗿

I know right, but that was not 100% clear for others IMHO 🗿

**Comment by proxafis** [comment]
any update on this?

**Comment by ThermicWaffle** [comment]
TVDB support was added in v3 (Seerr upgrade), could be worth revisiting?

**Comment by fallenbagel** [comment]
> TVDB support was added in v3 (Seerr upgrade), could be worth revisiting?

It's already being worked on.

**Comment by Yankees4life** [comment]
I'll cosign on this FR. I only have a finite amount of space so combining this feature with prefetcharr will help prevent my disk from being filled up

**Comment by Danjameswalker** [comment]
What's the latest update? Are we edging ever closer?

**Comment by fallenbagel** [comment]
> What's the latest update? Are we edging ever closer?

Yes. We literally have a pr for single episode availability open rn. After that merges, request comes next.

**Comment by Danjameswalker** [comment]
Amazing looking forward to it 😊

**Comment by slflowfoon** [comment]
Here is a concrete adjacent case where season-level re-requesting would help, without requiring an episode picker.

On Seerr 3.4.1, an existing show discovered through Jellyfin had a season marked `PARTIALLY_AVAILABLE` but no Seerr request. The season contained 10 episodes: episodes 1-7 were available, episode 8 had aired and was missing, and episodes 9-10 were still unaired. The request modal disabled the season, while `POST /api/v1/request` for that season returned `202 No seasons available to request`.

The season was already monitored in Sonarr, but the aired missing episode had not been acquired. This meant there was no way to put the season into Seerr’s request lifecycle or initiate a missing-episode search from Seerr.

Expected season-level behavior:

- A `PARTIALLY_AVAILABLE` season with no active request for the same quality tier remains requestable.
- Creating the request monitors the selected season and searches only its aired missing episodes.
- Future episodes remain monitored for RSS acquisition.
- The request remains approved while the season is partial and completes only when the full season is available.
- Existing active requests and `AVAILABLE`/`PROCESSING` seasons remain protected from duplicates.

I implemented and live-tested this narrower behavior in https://github.com/slflowfoon/seerr/commit/c20be039b8c158a21e4caa06e45c5f7f320ea82b. In the test, Seerr created one approved season request, sent Sonarr an `EpisodeSearch` containing only the missing aired episode ID, skipped the two future episodes, rejected a duplicate request, and retained the approved/partial request state.

This does not replace the episode-level availability/requesting work proposed here; it provides a season-level path for imported partial seasons in the meantime. Would the maintainers be open to this as a focused PR, possibly split into request eligibility and targeted Sonarr search changes?

Disclosure: the fork implementation and this investigation were developed with AI assistance and independently validated through project checks and a live integration test.

**Comment by fallenbagel** [comment]
> Here is a concrete adjacent case where season-level re-requesting would help, without requiring an episode picker.
> 
> On Seerr 3.4.1, an existing show discovered through Jellyfin had a season marked `PARTIALLY_AVAILABLE` but no Seerr request. The season contained 10 episodes: episodes 1-7 were available, episode 8 had aired and was missing, and episodes 9-10 were still unaired. The request modal disabled the season, while `POST /api/v1/request` for that season returned `202 No seasons available to request`.
> 
> The season was already monitored in Sonarr, but the aired missing episode had not been acquired. This meant there was no way to put the season into Seerr’s request lifecycle or initiate a missing-episode search from Seerr.
> 
> Expected season-level behavior:
> 
> - A `PARTIALLY_AVAILABLE` season with no active request for the same quality tier remains requestable.
> - Creating the request monitors the selected season and searches only its aired missing episodes.
> - Future episodes remain monitored for RSS acquisition.
> - The request remains approved while the season is partial and completes only when the full season is available.
> - Existing active requests and `AVAILABLE`/`PROCESSING` seasons remain protected from duplicates.
> 
> I implemented and live-tested this narrower behavior in https://github.com/slflowfoon/seerr/commit/c20be039b8c158a21e4caa06e45c5f7f320ea82b. In the test, Seerr created one approved season request, sent Sonarr an `EpisodeSearch` containing only the missing aired episode ID, skipped the two future episodes, rejected a duplicate request, and retained the approved/partial request state.
> 
> This does not replace the episode-level availability/requesting work proposed here; it provides a season-level path for imported partial seasons in the meantime. Would the maintainers be open to this as a focused PR, possibly split into request eligibility and targeted Sonarr search changes?
> 
> Disclosure: the fork implementation and this investigation were developed with AI assistance and independently validated through project checks and a live integration test.

AI-driven contributions aren't accepted here, per the [CONTRIBUTING.md AI policy](https://github.com/seerr-team/seerr/blob/970bb545716505e3d3d2fa7072a7aabbc5712c05/CONTRIBUTING.md#L9-L17). Saying it was "independently validated with live tests" doesn't change that. The analysis and the diff are still AI-authored as confirmed by you.

Season-level re-request isn't happening on the merits either. Once a season is requested, Sonarr is responsible for monitoring and searching for it. If Sonarr isn't grabbing an aired episode, that's a Sonarr-side issue, whether it's the indexer, quality profile, search, etc. That's not something Seerr should work around in its request lifecycle. Adding a separate season-request path that manually triggers "EpisodeSearch" for individual episodes would just duplicate what Sonarr already does and create another source of truth for monitoring state. We're not going to do that.

**This is also off-topic for this issue.** This thread is specifically about single-episode requesting, which already covers your "partially available season, one aired episode stuck" case once it ships. There's no need for a separate season-level path. If you want to propose something different, please open a separate issue for it. It doesn't belong in this thread.

**Comment by rrizik** [comment]
Just for tracking purposes, is it this PR? https://github.com/seerr-team/seerr/pull/3402
Which seems to be tied to two other PR's
- https://github.com/seerr-team/seerr/pull/3355
- https://github.com/seerr-team/seerr/pull/3361

So all three of these need to be merged for availability to be supported as a first step?

**Comment by fallenbagel** [comment]
> Just for tracking purposes, is it this PR? https://github.com/seerr-team/seerr/pull/3402
> Which seems to be tied to two other PR's
> - https://github.com/seerr-team/seerr/pull/3355
> - https://github.com/seerr-team/seerr/pull/3361
> 
> So all three of these need to be merged for availability to be supported as a first step?

Yes. All three will land together. Those are stacked PRs. So when you merge the upper most PR all the downstack PRs will merge with it.
